File: 03_MOBILE/ios/TerminalAttach/Resize/TerminalResizeCoordinator.swift
import Foundation

public struct TerminalViewport: Equatable {
  public let columns: Int
  public let rows: Int

  public init(columns: Int, rows: Int) {
    self.columns = columns
    self.rows = rows
  }
}

public final class TerminalResizeCoordinator {
  public private(set) var lastViewport: TerminalViewport?

  public init() {}

  public func update(columns: Int, rows: Int) -> TerminalViewport {
    let viewport = TerminalViewport(columns: columns, rows: rows)
    lastViewport = viewport
    return viewport
  }
}

File: 03_MOBILE/android/app/src/main/java/com/stratoc/mobile/terminal/resize/TerminalResizeCoordinator.kt
package com.stratoc.mobile.terminal.resize

data class TerminalViewport(val columns: Int, val rows: Int)

class TerminalResizeCoordinator {
    var lastViewport: TerminalViewport? = null
        private set

    fun update(columns: Int, rows: Int): TerminalViewport {
        return TerminalViewport(columns, rows).also { lastViewport = it }
    }
}
